tools/AutomationTestFramework/TestResources/Smoke.Helper.psm1

```diff
@@ -0,0 +1,110 @@
+﻿# ----------------------------------------------------------------------------------
+	#
+	# Copyright Microsoft Corporation
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	# http://www.apache.org/licenses/LICENSE-2.0
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+function LoginWithConnection([string] $connectionName, [string] $subscriptionName) {
+	
+	if ([string]::IsNullOrEmpty($connectionName)) {
+		$connectionName = 'AzureRunAsConnection';
+		Write-Output "Parameter 'connectionName' is not set - will use the default connection name $connectionName"
+	}
+	
+	try {
+		$servicePrincipalConnection = Get-AutomationConnection -Name $connectionName
+
+		"==> Logging in to Azure using connection $connectionName with subscription $subscriptionName..."
+		$null = Add-AzureRmAccount `
+			-ServicePrincipal `
+			-TenantId $servicePrincipalConnection.TenantId `
+			-ApplicationId $servicePrincipalConnection.ApplicationId `
+			-CertificateThumbprint $servicePrincipalConnection.CertificateThumbprint `
+			-Subscription $subscriptionName
+	} catch {
+		if (!$servicePrincipalConnection) {
+			$ErrorMessage = "Connection $connectionName not found."
+			throw $ErrorMessage
+		} else {
+			Write-Error -Message $_.Exception
+			throw $_.Exception
+		}
+	}
+}
+
+# https://blogs.endjin.com/2014/07/how-to-retry-commands-in-powershell/
+function Retry-Command
+{
+    param (
+		[ScriptBlock] $script,
+		[string] $connectionName, 
+		[string] $subscriptionName,
+        [int] $retyCount = 3,
```

Comment:
spacing

Reply:
Interesting. That wasn't like that in the original commit. Possibly a soft reset/merge issue?

Reply:
Are you using auto-format?  Often this occurs from mixing spaces and tabs.